tests/test_TFNetworkLayer.py

```diff
@@ -150,6 +150,137 @@ def test_concat_sources_missing_dim():
+          l.output.placeholder, [l.name, "shape:", tf.shape(l.output.placeholder)],
+          summarize=10, name="debug_print_layer_output_shape")
+      fetches.append(l.output.placeholder)
+    output = network.get_default_output_layer().output
+
+    tf.global_variables_initializer().run()
+    out = session.run(fetches=fetches,
+      feed_dict=feed_dict)
+    inp1, inp2 = network.layers["att_energy_dot"].sources[0].output.placeholder, network.layers["att_energy_dot"].sources[1].output.placeholder
+    res = network.layers["att_energy_dot"].output.placeholder
+    inp1, inp2, res = session.run([inp1, inp2, res], feed_dict=feed_dict)
+
+
+def test_CombineLayer_after_WindowLayer_and_SliceLayer():
+  import numpy as np
+  from TFNetwork import TFNetwork, ExternData
+  with make_scope() as session:
+    net = TFNetwork(extern_data=ExternData())
+    src_data = Data(name='t', shape=(None, 32, 128), dtype='float32', auto_create_placeholders=True)
+    window_size = 5
+    slice_size = window_size - 1
+
+    rnd = np.random.RandomState(42)
+    input_data = rnd.rand(10, 11, 32, 128)
+    seq_lens = np.array([11, 11, 11, 11, 11, 11, 11, 11, 11, 11])
+
+    with tf.variable_scope("src"):
+      src = InternalLayer(name="src", network=net, out_type={"dim": 128, "shape": (None, 32, 128),
+                                                             "batch_dim_axis": 0, "time_dim_axis": 1,
+                                                             "feature_dim_axis": 3, "sparse": False})
+      src.output = src_data
+      feed_dict = {src_data.placeholder: input_data,
+                   src_data.size_placeholder[0]: seq_lens}
+    with tf.variable_scope("win"):
+      win = WindowLayer(name="win", network=net, window_size=window_size, sources=[src], padding="valid",
+                        output=WindowLayer.get_out_data_from_opts(name="win", window_size=window_size,
+                                                                  padding="valid", sources=[src], network=net))
+    with tf.variable_scope("red"):
+      red = ReduceLayer(name="red", network=net, mode="AVG", axes="s:1", keep_dims=False, sources=[win],
+                        output=ReduceLayer.get_out_data_from_opts(name="red", network=net, mode="AVG",
+                                                                  axes="s:1", keep_dims=False, sources=[win]))
+    with tf.variable_scope("slice"):
+      sl = SliceLayer(name="slice", axis="t", slice_start=slice_size // 2, slice_end=-(slice_size // 2),
+                      sources=[src], network=net,
+                      output=SliceLayer.get_out_data_from_opts(name="slice", axis="t", slice_start=slice_size // 2,
+                                                               slice_end=-(slice_size // 2), sources=[src],
+                                                               network=net))
+    with tf.variable_scope("combine"):
+      combine = CombineLayer(name="combine", network=net, kind="add", sources=[sl, red],
```

Comment:
Please add some print statements, for debugging, such as:
```
print("sl:", sl)
print("red:", red)
print("combine:", combine)
```

Reply:
Done